### PR TITLE
Changed: Provider and Terraform versions

### PR DESCRIPTION
**Pull request by xescugc**
Basically updated the versions of the Provieders and TF.

TF: 0.13.5-cy
AWS: 3.1.0
AZ: 2.34.0
GCP: 3.40.0

We cannot update to the latest version as they require the TF-SDK version v2 which is not compatible with the code we have right now.

I'll work to solve that problem too.

### CODE REVIEW

**Comment by xescugc** [comment]
Ok ready to review.

File: state/writer_test.go [hunk 1]
```diff
@@ -168,12 +168,12 @@ func TestSync(t *testing.T) {
          ],
          "mode":"managed",
          "name":"name",
-         "provider":"provider.aws",
+      	 "provider": "provider[\"registry.terraform.io/hashicorp/aws\"]",
```
Comment (kerak19) [inline state/writer_test.go:171]: I think the formatting is wrong (or GH)

File: tag/tag.go [hunk 2]
```diff
@@ -2,14 +2,18 @@ package tag
...
 	"github.com/aws/aws-sdk-go/aws"
 	"github.com/aws/aws-sdk-go/service/ec2"
 	"github.com/chr4/pwgen"
 	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
-	"github.com/hashicorp/terraform/config"
 )
 
+// NameRegexp is the new regexp used to validate the names
+// of the resources on TF (defined on configs/configschema/internal_validate.go)
+var NameRegexp = regexp.MustCompile(`^[a-z0-9_]+$`)
```
Comment (kerak19) [inline tag/tag.go:15]: I think it should be private

File: tag/tag.go [hunk 3]
```diff
@@ -48,5 +52,5 @@ func GetNameFromTag(key string, srd *schema.ResourceData, fallback string) strin
 // isValidResourceName checks with the TF regex
 // for names to validate if it's valid
 func isValidResourceName(name string) bool {
-	return config.NameRegexp.Match([]byte(name))
+	return NameRegexp.Match([]byte(name))
```
Comment (kerak19) [inline tag/tag.go:55]: `MatchString(name)`

**Review by kerak19 — CHANGES_REQUESTED** [review]

File: state/writer_test.go (hunk 1 above)
Comment (xescugc) [reply state/writer_test.go:171]: I think it's GH, I don't see anything wrong hehe.

Comment (kerak19) [reply state/writer_test.go:171]: image
If it's GH then we can resolve this

**Comment by xescugc** [comment]
Done

**Comment by kerak19** [comment]
RS

**Review by kerak19 — APPROVED** [review]

**Review by tormath1 — APPROVED** [review]

**Pull request merged**